src/shared/hocs/withLoading.tsx

```diff
@@ -0,0 +1,15 @@
+import React, { ComponentType, ReactElement } from 'react';
+
+import { Spinner } from '@viaa/avo2-components';
+
+interface WithLoadingProps {
+	loading: boolean;
+}
+
+const withLoading = (WrappedComponent: ComponentType<any>): ComponentType<any> => (
```

Comment:
In the future this will be handled by the apollo graphql client
https://github.com/viaacode/avo2-client/blob/c8e6152af3798d7c02268c00b19649c0d00daaf6/src/shared/components/DataComponent/DataQueryComponent.tsx#L28

after this PR is merged: https://github.com/viaacode/avo2-client/pull/47